segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -246,6 +249,80 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
+     * @param metadataSegments  A map of back of metadata segments along with their container Ids.
+     * @param containersMap     A map of {@link DebugStreamSegmentContainer} instances with their container Ids.
+     * @param executorService   A thread pool for execution.
+     * @throws Exception        If an exception occurred. This could be one of the following:
+     *                              * TimeoutException:     If the calls for computation(used in the method)
+     *                                                      didn't complete in time.
+     *                              * IOException     :     If a general IO exception occurred.
+     */
+    public static void updateCoreAttributes(Map<Integer, String> metadataSegments,
+                                            Map<Integer, DebugStreamSegmentContainer> containersMap,
+                                            ExecutorService executorService) throws Exception {
+        val args = IteratorArgs.builder().fetchTimeout(TIMEOUT).build();
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containersMap.size());
+
+        ArrayList<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (val metadataEntry : metadataSegments.entrySet()) {
+            val containerForSegmentsContained = containersMap.get(metadataEntry.getKey());
+            val containerForBackUpMetadataSegment = containersMap.get(segToConMapper.getContainerId(metadataEntry.getValue()));
+            log.info("container id: {}, segment Name: {}", containerForBackUpMetadataSegment.getId(), metadataEntry.getValue());
+            val tableExtension = containerForBackUpMetadataSegment.getExtension(ContainerTableExtension.class);
+            val entryIterator = tableExtension.entryIterator(metadataEntry.getValue(), args)
+                    .get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+
+            val metadataSegmentName = NameUtils.getMetadataSegmentName(containerForSegmentsContained.getId());
+            // Store the segments in a set
+            entryIterator.forEachRemaining(item -> {
+                for (val entry : item.getEntries()) {
+                    val segmentInfo = MetadataStore.SegmentInfo.deserialize(entry.getValue());
+                    val properties = segmentInfo.getProperties();
+
+                    // skip, if this is a metadata segment
+                    if (metadataSegmentName.equals(properties.getName())) {
+                        continue;
+                    }
+
+                    List<AttributeUpdate> attributeUpdates = properties.getAttributes().entrySet().stream()
+                            .map(e -> new AttributeUpdate(e.getKey(), AttributeUpdateType.Replace, e.getValue()))
+                            .collect(Collectors.toList());
+                    log.info("Updating attributes = {} for segment '{}'", attributeUpdates, properties.getName());
```

Comment:
"Segment Name: {}, Attribute Updates: {}`

(put attribute updates second since it's going to be a long list)

Reply:
Fixed.